logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/auth/LoginUseCase.kt

```diff
@@ -133,16 +136,23 @@ internal class LoginUseCaseImpl internal constructor(
             .fold({
                 when (it) {
                     is NetworkFailure.ProxyError -> AuthenticationResult.Failure.SocketError
-                    is NetworkFailure.ServerMiscommunication -> handleServerMiscommunication(it)
+                    is NetworkFailure.ServerMiscommunication -> handleServerMiscommunication(it, isEmail, cleanUserIdentifier)
                     is NetworkFailure.NoNetworkConnection, NetworkFailure.FederatedBackendFailure ->
                         AuthenticationResult.Failure.Generic(it)
                 }
             }, {
+                if (isEmail && secondFactorVerificationCode != null) {
+                    secondFactorVerificationRepository.storeVerificationCode(cleanUserIdentifier, secondFactorVerificationCode)
```

Comment:
Why is the 2fa only stored when logging in with e-email? Shouldn't we re-use the 2fa code also when logging in with the handle?

The get back the self user when logging in so we should have all the data, no?

Reply:
Ok, strange. micro services something something